### DISCUSSION TITLE
How to copy a file to Clipboard?

### DISCUSSION THREAD

**Discussion by ionite34** [AvaloniaUI/Avalonia#12840] [Q&A]
There doesn't seem to be documentation on how to copy a local file to the clipboard. I tried the following but it copies something that is not able to be pasted on Windows.
```cs
var data = new DataObject();
data.Set(DataFormats.Files, "some/path");
await Clipboard.SetDataAsync(DataFormats.Files, data);
```

When copying a file on Windows and using `Clipboard.GetDataAsync` I get an enumerable of `BclStorageFile`, but this is an internal class. What am I supposed to pass as an argument to SetDataAsync to set a file?

**Comment by timunie** [comment]
Clipboard needs some refractoring. See #11011 . For now, I use access to the WPF Clipboard on Windows. Luckily I don't have to target other OS, so that works for me. Don't know if it will work for you.

**Comment by Koichi-hub** [comment]
Hello stranger! This discussion was closed, but i have solution for you
Over here:
```c#
if (Application.Current?.ApplicationLifetime is not IClassicDesktopStyleApplicationLifetime desktop ||
    desktop.MainWindow?.Clipboard is not { } clipboard || desktop.MainWindow?.StorageProvider is not { } storageProvider)
{
    return;
}

IStorageFile? file = await storageProvider.TryGetFileFromPathAsync("/some/path/to/file");
if (file != null)
{
    DataObject dataObject = new();
    dataObject.Set(DataFormats.Files, new List<IStorageFile> { file });
    await clipboard.SetDataObjectAsync(dataObject);
}
```

It was worked for me. I found this line `storageProvider.TryGetFileFromPathAsync` [here](https://github.com/AvaloniaUI/Avalonia/blob/b16975cb0402cd10358f17135d25c3dd3aac7ac6/samples/ControlCatalog/Pages/ClipboardPage.xaml.cs#L90)

This code let you copy file by path in clipboard
